qiita_pet/handlers/qiita_redbiom.py

```diff
@@ -83,42 +84,81 @@ def _redbiom_search(self, query, search_on, callback):
+                                name, command_id,
+                                (main_query.children).artifact_id AS
+                                    artifact_id
+                            FROM main_query
+                            JOIN qiita.artifact a ON
+                                (main_query.children).artifact_id =
+                                    a.artifact_id
+                            JOIN qiita.artifact_type at ON (
+                                at.artifact_type_id = a.artifact_type_id
+                                AND artifact_type = 'BIOM')),
+                         parent_query AS (
+                            SELECT artifact_query.*,
+                                array_agg(parent_params) as parent_parameters
+                            FROM artifact_query
+                            LEFT JOIN qiita.parent_artifact pa ON (
+                                artifact_query.artifact_id = pa.artifact_id)
+                            LEFT JOIN qiita.artifact a ON (
+                                pa.parent_id = a.artifact_id),
+                                json_each_text(command_parameters)
+                                    parent_params
+                            GROUP BY artifact_query.study_title,
+                                artifact_query.study_id,
+                                artifact_query.samples, artifact_query.name,
+                                artifact_query.command_id,
+                                artifact_query.artifact_id)
+                        SELECT * FROM parent_query
+                        ORDER BY parent_parameters, artifact_id
                         """
+
+                        sql_params = """
+                        SELECT parameter_set_name, array_agg(ps) AS param_set
+                        FROM qiita.default_parameter_set,
+                            json_each_text(parameter_set) ps
+                        GROUP BY parameter_set_name"""
+
                         with qdbsc.TRN:
-                            qdbsc.TRN.add(sql, [tuple(features)])
                             results = []
                             commands = {}
+                            # obtaining all existing parameters, note that
+                            # they are not that many (~40) and we don't expect
+                            # to have a huge growth in the near future
+                            qdbsc.TRN.add(sql_params)
+                            params = {pname: eval(params) for pname, params
+                                      in qdbsc.TRN.execute_fetchindex()}
+
+                            # now let's get the actual artifacts
+                            qdbsc.TRN.add(sql, [tuple(features)])
                             for row in qdbsc.TRN.execute_fetchindex():
-                                title, sid, samples, name, cid, aid = row
+                                title, sid, samples, name, cid, aid, pp = row
```

Comment:
I dont understand how the select on `main_query` did not change but somehow `row` has an additional element now?